internal/e2e/basic_test.go

```diff
@@ -0,0 +1,37 @@
+//go:build e2e
+
+package e2e
+
+import (
+	"syscall"
+	"testing"
+)
+
+func TestBasicE2E(t *testing.T) {
+	ctx := t.Context()
+	tc := newTestCase(t, `
+agents:
+  queue: {{.queue}}
+steps:
+  - command: echo hello world
+`)
+
+	agent := tc.startAgent()
+	build := tc.triggerBuild()
+	state, err := tc.waitForBuild(ctx, build)
+	if err != nil {
+		t.Fatalf("tc.waitForBuild(build %s) error = %v", build.ID, err)
+	}
+	if got, want := state, "passed"; got != want {
+		t.Errorf("Build state = %q, want %q", got, want)
+	}
+
+	// TODO: add ability to inspect job logs
+
+	if err := agent.Process.Signal(syscall.SIGTERM); err != nil {
+		t.Errorf("agent.Process.Signal(%d) = %v", syscall.SIGTERM, err)
+	}
+	if err := agent.Wait(); err != nil {
+		t.Errorf("agent.Wait() = %v", err)
+	}
```

Comment:
this should get handled by the test completing, right? when the test finishes, its context gets cancelled, killing the process?

Reply:
We catch `SIGINT`, `SIGTERM`, `SIGQUIT`, and `SIGHUP`. We're not allowed to catch `SIGKILL`. The difference between `SIGQUIT` and `SIGTERM` is that we proceed directly to cancelling jobs with `SIGQUIT`, but that only happens for the second `SIGTERM`.

https://github.com/buildkite/agent/blob/c1039ae975a77006442352a1731a63e3f8df97d5/clicommand/agent_start.go#L1385-L1456